nucypher/characters/control/interfaces.py

```diff
@@ -99,7 +100,21 @@ def grant(self,
         response_data = {'treasure_map': new_policy.treasure_map,
                          'policy_encrypting_key': new_policy.public_key,
                          'alice_verifying_key': new_policy.alice.stamp}
+        return response_data
+
+    def revoke(self, policy_encrypting_key):
+        policy = self.character.active_policies[policy_encrypting_key]
+
+        failed_revocations = self.character.revoke(policy)
+        if len(failed_revocations) > 0:
+            for node_id, attempt in failed_revocations.items():
+                revocation, fail_reason = attempt
+                if fail_reason == NotFound:
+                    del(failed_revocations[node_id])
+        elif len(failed_revocations) <= (policy.n - policy.treasure_map.m + 1):
```

Comment:
Wouldn't this condition always be satisfied though?
E.g. the first `if` makes it so that we get here only if `len(failed_revocations) == 0`, which is (of course!) always `<= n - m + 1`.

Reply:
Perhaps @tuxxy didn't mean an `elif` but an `if`?

```suggestion
        if len(failed_revocations) <= (policy.n - policy.treasure_map.m + 1):
```
So the `if` in L109 is first cleaning the list of failed revocations, and then this line comes after that once the list only contains revocations with a valid reason? @tuxxy

Reply:
@cygnusv Yeah, this is a good catch.